aleph_scoring/metrics/__init__.py

```diff
@@ -207,6 +210,55 @@ def get_ipv4(url: str) -> Optional[str]:
         return None
 
 
+def get_ipv6(url: str) -> Optional[str]:
+    domain = get_url_domain(url)
+    try:
+        addrinfo = socket.getaddrinfo(domain, None, socket.AF_INET6)
+        return addrinfo[0][4][0]
+    except socket.gaierror:
+        return None
+
+
+def get_executable_ipv6(
+    crn_ipv6_range: IPv6Network, vm_type: VmType, item_hash: str
+) -> IPv6Address:
+    ipv6_elems = crn_ipv6_range.exploded.split(":")[:4]
+    ipv6_elems += [str(vm_type.value)]
+
+    # Add the item hash of the VM as the last 44 bits of the IPv6 address.
+    # We expect the VM interface to be set up to use the "1" address of the /124 subnet.
+    ipv6_elems += [item_hash[0:4], item_hash[4:8], item_hash[8:11] + "1"]
+
+    return IPv6Address(":".join(ipv6_elems))
+
+
+def ping(ip_address: Union[IPv4Address, IPv6Address]) -> bool:
+    try:
+        result = subprocess.run(["ping", "-c", "1", str(ip_address)], timeout=2)
```

Comment:
Note: Using `icmplib` will be possible once Debian 11 is deprecated.
```suggestion
        # Can be replaced with `icmplib` on Debian 12 and Ubuntu 22.04
        # https://packages.debian.org/bookworm/python3-icmplib
        # https://packages.ubuntu.com/jammy/python3-icmplib
        result = subprocess.run(["ping", "-c", "1", str(ip_address)], timeout=2)
```